### PR TITLE
Tsk1031: Update to Spring Boot 2.2.4 and Spring 5.2.3

### PR DESCRIPTION
**Pull request by holgerhagen**

### CODE REVIEW

File: history/taskana-simplehistory-rest-spring-example/pom.xml [hunk 1]
```diff
@@ -1,99 +1,111 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-rest-spring-example/pom.xml]: We don't have a formatter yet for our *.xml files. So this is fine for now.

What did you exactly change?

File: history/taskana-simplehistory-spring-test/pom.xml [hunk 2]
```diff
@@ -1,175 +1,187 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-spring-test/pom.xml]: Same as above.. Fine for now. What did you exactly change?

File: lib/taskana-core/src/main/java/pro/taskana/common/internal/jobs/WorkbasketCleanupJob.java [hunk 3]
```diff
@@ -1,149 +1,146 @@
-package pro.taskana.common.internal.jobs;
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/common/internal/jobs/WorkbasketCleanupJob.java]: Why is the diff so big?

Comment (mustaphazorgati) [reply lib/taskana-core/src/main/java/pro/taskana/common/internal/jobs/WorkbasketCleanupJob.java]: Please reformat.

File: lib/taskana-spring-example/src/test/java/pro/taskana/TaskanaTransactionIntTest.java [hunk 4]
```diff
@@ -45,19 +40,20 @@
...
 import pro.taskana.workbasket.api.exceptions.WorkbasketNotFoundException;
 import pro.taskana.workbasket.internal.WorkbasketImpl;
 
 /** Test for internal transaction management. */
 @ExtendWith(SpringExtension.class)
 @SpringBootTest(
     classes = TaskanaConfigTestApplication.class,
-    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = "spring.main.allow-bean-definition-overriding=true")
```
Comment (mustaphazorgati) [inline lib/taskana-spring-example/src/test/java/pro/taskana/TaskanaTransactionIntTest.java]: Why did you add this property?

File: pom.xml [hunk 5]
```diff
@@ -1,351 +1,352 @@
-<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (mustaphazorgati) [inline pom.xml]: Fine for now. Anything else except for the version changes?

File: rest/taskana-rest-spring-example-boot/pom.xml [hunk 6]
```diff
@@ -1,201 +1,213 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-boot/pom.xml]: What changed?

File: rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/rest/ExampleRestApplication.java [hunk 7]
```diff
@@ -1,74 +1,73 @@
-package pro.taskana.rest;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-boot/src/main/java/pro/taskana/rest/ExampleRestApplication.java]: Please reformat.

File: lib/taskana-spring-example/src/test/java/pro/taskana/TaskanaTransactionIntTest.java (hunk 4 above)
Comment (mustaphazorgati) [reply lib/taskana-spring-example/src/test/java/pro/taskana/TaskanaTransactionIntTest.java]: And why is it here and not in the application.properties file?

File: rest/taskana-rest-spring-example-boot/src/main/resources/application.properties [hunk 8]
```diff
@@ -1,4 +1,4 @@
-logging.level.pro.taskana=INFO
+logging.level.pro.taskana=DEBUG
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-boot/src/main/resources/application.properties]: do we need debug here?

File: rest/taskana-rest-spring-example-common/pom.xml [hunk 9]
```diff
@@ -1,136 +1,128 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-common/pom.xml]: what changed?

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/RestHelper.java [hunk 10]
```diff
@@ -1,70 +1,80 @@
-package pro.taskana;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/RestHelper.java]: Please reformat

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/jobs/AsyncUpdateJobIntTest.java [hunk 11]
```diff
@@ -1,168 +1,168 @@
...
-  }
-
-  private void verifyTaskIsModifiedAfterOrEquals(String taskId, Instant before)
-      throws InvalidArgumentException {
-
-    ResponseEntity<TaskResource> taskResponse =
-        template.exchange(
-            restHelper.toUrl(Mapping.URL_TASKS_ID, taskId),
-            HttpMethod.GET,
-            new HttpEntity<>(restHelper.getHeadersAdmin()),
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/jobs/AsyncUpdateJobIntTest.java]: Please reformat.

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/ldap/LdapTest.java [hunk 12]
```diff
@@ -1,36 +1,37 @@
-package pro.taskana.ldap;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/ldap/LdapTest.java]: Please reformat

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/rest/AccessIdValidationControllerIntTest.java [hunk 13]
```diff
@@ -1,86 +0,0 @@
-package pro.taskana.rest;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/rest/AccessIdValidationControllerIntTest.java]: Where did these tests go?

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/rest/ExampleRestApplication.java [hunk 14]
```diff
@@ -1,74 +1,73 @@
-package pro.taskana.rest;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/rest/ExampleRestApplication.java]: Please reformat.

File: rest/taskana-rest-spring-example-wildfly/pom.xml [hunk 15]
```diff
@@ -1,359 +1,355 @@
-<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/pom.xml]: what changed?

File: rest/taskana-rest-spring-example-wildfly/src/main/java/pro/taskana/TaskanaWildFlyApplication.java [hunk 16]
```diff
@@ -1,122 +1,121 @@
-package pro.taskana;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/src/main/java/pro/taskana/TaskanaWildFlyApplication.java]: Please reformat

File: rest/taskana-rest-spring-example-wildfly/src/main/java/pro/taskana/wildfly/security/WildflyWebSecurityConfig.java [hunk 17]
```diff
@@ -1,203 +1,203 @@
-package pro.taskana.wildfly.security;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/src/main/java/pro/taskana/wildfly/security/WildflyWebSecurityConfig.java]: Please reformat

File: rest/taskana-rest-spring-example-wildfly/src/main/resources/application.properties [hunk 18]
```diff
@@ -1,41 +1,44 @@
-spring.profiles.active=@activatedProperties@
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring-example-wildfly/src/main/resources/application.properties]: Please run `git config --global core.autocrlf input` so that the line endings are unix and not windows :)

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java [hunk 19]
```diff
@@ -1,324 +1,325 @@
-package pro.taskana.rest;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java]: Please reformat

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/RestConfiguration.java [hunk 20]
```diff
@@ -1,77 +1,78 @@
-package pro.taskana.rest;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/RestConfiguration.java]: Please reformat.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketController.java [hunk 21]
```diff
@@ -82,8 +83,7 @@
 
   private WorkbasketAccessItemResourceAssembler workbasketAccessItemResourceAssembler;
 
-  WorkbasketController(
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketController.java]: This should not have happened. Reformat please :)

File: pom.xml [hunk 22]
```diff
@@ -50,13 +50,14 @@
...
+		<version.spring>5.2.3.RELEASE</version.spring>
+        <version.spring.security>5.2.1.RELEASE</version.spring.security>
 		<version.spring.core>2.0.0.RELEASE</version.spring.core>
-		<version.spring.boot>2.0.2.RELEASE</version.spring.boot>
+		<version.spring.boot>2.2.4.RELEASE</version.spring.boot>
 		<version.spring.restdocs>2.0.2.RELEASE</version.spring.restdocs>
 		<version.spring.mybatis>2.0.3</version.spring.mybatis>
-		<version.spring.ldap>2.3.2.RELEASE</version.spring.ldap>
 		<version.spring.hateos>0.24.0.RELEASE</version.spring.hateos>
+        <version.spring.ldap>2.3.2.RELEASE</version.spring.ldap>
```
Comment (gitgoodjhe) [inline pom.xml]: pls adjust this line

File: pom.xml [hunk 23]
```diff
@@ -50,13 +50,14 @@
 		<version.sonatype>1.6.8</version.sonatype>
 
 		<!-- spring dependencies -->
-		<version.spring>5.0.5.RELEASE</version.spring>
+		<version.spring>5.2.3.RELEASE</version.spring>
+        <version.spring.security>5.2.1.RELEASE</version.spring.security>
```
Comment (gitgoodjhe) [inline pom.xml]: pls adjust this line

File: history/taskana-simplehistory-spring-test/pom.xml [hunk 24]
```diff
@@ -32,7 +32,18 @@
...
+                    <groupId>org.springframework.hateoas</groupId>
+                    <artifactId>spring-hateoas</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.hateoas</groupId>
+            <artifactId>spring-hateoas</artifactId>
+            <version>${version.spring.hateos}</version>
+        </dependency>
```
Comment (gitgoodjhe) [inline history/taskana-simplehistory-spring-test/pom.xml]: Pls adjust identation levels

File: rest/taskana-rest-spring-example-boot/src/main/resources/application.properties (hunk 8 above)
Comment (gitgoodjhe) [reply rest/taskana-rest-spring-example-boot/src/main/resources/application.properties]: Still open

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java [hunk 25]
```diff
@@ -68,15 +76,11 @@ void testGetAllClassificationsKeepingFilters() {
...
-            .getBody()
-            .getLink(Link.REL_SELF)
-            .getHref()
-            .endsWith("/api/v1/classifications?domain=DOMAIN_A&sort-by=key&order=asc"));
-    assertEquals(17, response.getBody().getContent().size());
-    assertEquals("A12", response.getBody().getContent().iterator().next().key);
+    assertThat(response.getBody().getLink(Link.REL_SELF)).isNotNull();
+    assertThat(response.getBody().getLink(Link.REL_SELF).getHref())
+        .endsWith("/api/v1/classifications?domain=DOMAIN_A&sort-by=key&order=asc");
+    assertThat(response.getBody().getContent().size()).isEqualTo(17);
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java]: assertThat(response.getBody().getContent()).hasSize(17)

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java [hunk 26]
```diff
@@ -36,7 +39,7 @@ void testQueryGroupsByDn() {
             HttpMethod.GET,
             restHelper.defaultRequest(),
             ParameterizedTypeReference.forType(List.class));
-    assertEquals(1, response.getBody().size());
+    assertThat(response.getBody().size()).isEqualTo(1);
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java]: hasSize()

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java [hunk 27]
```diff
@@ -47,6 +50,42 @@ void testQueryGroupsByCn() {
...
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java]: hasSize()

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/ldap/LdapTest.java [hunk 28]
```diff
@@ -30,7 +30,7 @@ void testFindUsers() throws InvalidArgumentException {
     if (ldapClient.useLdap()) {
       List<AccessIdResource> usersAndGroups = ldapClient.searchUsersAndGroups("ser0");
       System.out.println("#### found " + LoggerUtils.listToString(usersAndGroups));
-      assertEquals(50, usersAndGroups.size());
+      assertThat(usersAndGroups.size()).isEqualTo(50);
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/ldap/LdapTest.java]: hasSize()

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java [hunk 29]
```diff
@@ -47,6 +50,42 @@ void testQueryGroupsByCn() {
...
+    ResponseEntity<List<AccessIdResource>> response =
+        template.exchange(
+            restHelper.toUrl(Mapping.URL_ACCESSID) + "?search-for=ali",
+            HttpMethod.GET,
+            restHelper.defaultRequest(),
+            ParameterizedTypeReference.forType(AccessIdListResource.class));
+
+    List<AccessIdResource> body = response.getBody();
+    assertThat(body).isNotNull();
+    assertThat(body.size()).isEqualTo(3);
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java]: hasSize()

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java [hunk 30]
```diff
@@ -55,8 +63,8 @@ void testGetAllClassificationsFilterByCustomAttribute() {
             HttpMethod.GET,
             restHelper.defaultRequest(),
             ParameterizedTypeReference.forType(ClassificationSummaryListResource.class));
-    assertNotNull(response.getBody().getLink(Link.REL_SELF));
-    assertEquals(13, response.getBody().getContent().size());
+    assertThat(response.getBody().getLink(Link.REL_SELF)).isNotNull();
+    assertThat(response.getBody().getContent().size()).isEqualTo(13);
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java]: hasSize()

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java [hunk 31]
```diff
@@ -88,21 +92,17 @@ void testGetSecondPageSortedByKey() {
...
-            .getLink(Link.REL_SELF)
-            .getHref()
-            .endsWith(
-                "/api/v1/classifications?"
-                    + "domain=DOMAIN_A&sort-by=key&order=asc&page=2&page-size=5"));
-    assertNotNull(response.getBody().getLink(Link.REL_FIRST));
-    assertNotNull(response.getBody().getLink(Link.REL_LAST));
-    assertNotNull(response.getBody().getLink(Link.REL_NEXT));
-    assertNotNull(response.getBody().getLink(Link.REL_PREVIOUS));
+    assertThat(response.getBody().getContent().size()).isEqualTo(5);
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java]: hasSize()

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java [hunk 32]
```diff
@@ -213,52 +212,46 @@ void testCreateClassificationWithParentKeyInDomain_aShouldCreateAClassificationI
...
+    
+    assertThatThrownBy(
+        () ->
+        template.exchange(
+            restHelper.toUrl(Mapping.URL_CLASSIFICATIONS),
+            HttpMethod.POST,
+            new HttpEntity<>(newClassification, restHelper.getHeaders()),
+            ParameterizedTypeReference.forType(ClassificationResource.class)))
+        .isInstanceOf(HttpClientErrorException.class)
+        .hasFieldOrPropertyWithValue("statusCode", HttpStatus.BAD_REQUEST);
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/ClassificationControllerIntTest.java]: Pls use extracting method to have a consistent approach with the other tests

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java [hunk 33]
```diff
@@ -47,6 +50,42 @@ void testQueryGroupsByCn() {
...
+    assertThat(body)
+        .extracting(AccessIdResource::getName)
+        .containsExactlyInAnyOrder("Tralisch, Thea", "Bert, Ali", "Mente, Ali");
+  }
+
+  @Test
+  void testBadRequestWhenSearchForIsTooShort() {
+    try {
+      template.exchange(
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/AccessIdControllerIntTest.java]: Pls use the assertThatThrownBy-pattern to be consistent with other tests
